pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -57,12 +61,26 @@ public Renderer getRenderer() {
     }
 
     public void start(String baseDir) {
+
+        Properties properties = createProperties();
+        
+        Charset charset;
+        {
+            String s = (String) properties.get("encoding");
+            if (null == s) {
+                charset = StandardCharsets.UTF_8;
+                properties.put("encoding", charset.name());
+            } else {
+                charset = Charset.forName(s);
+            }
+        }
+
         try {
             if (toConsole) {
-                writer = new BufferedWriter(new OutputStreamWriter(System.out));
+                writer = new BufferedWriter(new OutputStreamWriter(System.out, charset));
```

Comment:
The only "cross-jre" workaround I've been able to find so far is [this one](http://hg.openjdk.java.net/code-tools/jmh/file/0797222e066e/jmh-core/src/main/java/org/openjdk/jmh/util/Utils.java#l173). Being used by JMH makes me think it's pretty solid and covers pretty much all JREs. I'll keep looking into this tomorrow.